geode-core/src/integrationTest/java/org/apache/geode/distributed/LocatorIntegrationTest.java

```diff
@@ -63,162 +64,168 @@
-    File locatorFile = new File("locator" + portNumber + "view.dat");
-    if (locatorFile.exists()) {
-      locatorFile.delete();
-    }
-  }
-
   @After
   public void tearDown() {
+    JGroupsMessenger.THROW_EXCEPTION_ON_START_HOOK = false;
     if (locator != null) {
       locator.stop();
     }
-    assertEquals(false, Locator.hasLocator());
+    assertThat(Locator.hasLocator()).isFalse();
   }
 
   /**
-   * GEODE-4176: locator creates "locator0view.dat" file when started with port 0
-   *
+   * Fix: locator creates "locator0view.dat" file when started with port 0
    */
   @Test
   public void testThatLocatorDoesNotCreateFileWithZeroPort() throws Exception {
     deleteLocatorViewFile(0);
-    Properties dsprops = new Properties();
-    dsprops.setProperty(MCAST_PORT, "0");
-    dsprops.setProperty(ENABLE_CLUSTER_CONFIGURATION, "false");
-    dsprops.setProperty(LOCATOR_WAIT_TIME, "1"); // seconds
-    dsprops.setProperty(LOG_FILE, "");
-    locator = Locator.startLocatorAndDS(port, null, dsprops);
+
+    Properties configProperties = new Properties();
+    configProperties.setProperty(MCAST_PORT, "0");
+    configProperties.setProperty(ENABLE_CLUSTER_CONFIGURATION, "false");
+    configProperties.setProperty(LOCATOR_WAIT_TIME, "1"); // seconds
+    configProperties.setProperty(LOG_FILE, "");
+
+    locator = Locator.startLocatorAndDS(port, null, configProperties);
+
     File viewFile = new File("locator0view.dat");
-    assertFalse("file should not exist: " + viewFile.getAbsolutePath(), viewFile.exists());
+
+    assertThat(viewFile).doesNotExist();
   }
 
   /**
-   * TRAC #45804: if jmx-manager-start is true in a locator then gfsh connect will fail
+   * Fix: if jmx-manager-start is true in a locator then gfsh connect will fail
    */
   @Test
   public void testGfshConnectShouldSucceedIfJmxManagerStartIsTrueInLocator() throws Exception {
-    Properties dsprops = new Properties();
     int jmxPort = getRandomAvailablePort(SOCKET);
-    dsprops.setProperty(MCAST_PORT, "0");
-    dsprops.setProperty(JMX_MANAGER_PORT, "" + jmxPort);
-    dsprops.setProperty(JMX_MANAGER_START, "true");
-    dsprops.setProperty(JMX_MANAGER_HTTP_PORT, "0");
-    dsprops.setProperty(ENABLE_CLUSTER_CONFIGURATION, "false");
-    dsprops.setProperty(LOG_FILE, "");
-    System.setProperty(DistributionConfig.GEMFIRE_PREFIX + "disableManagement", "false"); // not
-                                                                                          // needed
-    try {
-      locator = Locator.startLocatorAndDS(port, null, dsprops);
-      List<JmxManagerProfile> alreadyManaging =
-          GemFireCacheImpl.getInstance().getJmxManagerAdvisor().adviseAlreadyManaging();
-      assertEquals(1, alreadyManaging.size());
-      assertEquals(GemFireCacheImpl.getInstance().getMyId(),
-          alreadyManaging.get(0).getDistributedMember());
-    } finally {
-      System.clearProperty(DistributionConfig.GEMFIRE_PREFIX + "enabledManagement");
```

Comment:
Is it correct to leave this System variable set after the test completes?

Reply:
Adding RestoreSystemProperties JUnit Rule restores all System Properties after each test.